Editorial pass to draft-ietf-ccwg-rfc5033bis.md (#83)
ietf-wg-ccwg/rfc5033bis: draft-ietf-ccwg-rfc5033bis.md

this space.

This document is meant to reduce the barriers to entry for new congestion
control work to the IETF. As such, proponents ought not to interpret these criteria as a
checklist of requirements before approaching the IETF. Instead, proponents
are encouraged to think about these issues beforehand, and have the willingness
to do the work implied by the remainder of this document.

# Document Status

This document applies to proposals that seek Experimental or
Standards Track status. Evaluation of both cases involves the same questions,
but with different expectations for both the answers and the degree of
certainty it the answers.

Congestion control algorithms without experience of Internet-scale deployment SHOULD
seek Experimental status until real-world data is able to answer the questions
in {{general-use}}. Congestion control algorithms with a record of measured Internet-
scale deployment MAY directly seek the Standards Track if the community believes
it is safe, and the design is stable, guided by the considerations in

> {{general-use}}. The exince of this data does not waive the other

Reviewer: ```suggestion
{{general-use}}. The existence of this data does not waive the other
```

considerations in this document.

Algorithms that are designed for special environments (e.g., data centers) and

> forbidden from use in the Internet would, of course, inad seek real-world

Reviewer: ```suggestion
forbidden from use in the Internet would, of course, instead seek real-world
```

In contexts where differing congestion control
algorithms are used, it is important to understand whether
a proposal can induce more
harm to sharing flows than for the existing

> defined methods. The measure of harm is not restricted to the equality

Reviewer 2: This does not read well: "In contexts where differing congestion control
algorithms are used, it is important to understand whether
a proposal can induce more
harm to sharing flows than for the existing
defined methods." 

Proposal: "In contexts where differing congestion control
algorithms are used, it is important to understand whether
a proposal can induce more
harm than previously defined algorithms to flows sharing a common bottleneck."

Author: better as far as I can see
but ... Looking at this, i worry about whether "can" could be misread?
 /can/could result in/ ?

Reviewer 2: You are right, but I think this may be OBE. I was commenting on an old PR. The whole text has been reformatted already.

Reviewer 2: I will open a specific PR.

A proposals for a new congestion control algorithm MUST consider how new and short-lived flows affect long-lived flows,
and vice versa.

## Mixed Algorithm Behavior

These criteria evaluate the interaction of the proposal with commonly deployed
congestion control algorithms.

### Existing General-Purpose Transports

Evaluate the impact on TCP {{!RFC9293}}, SCTP {{!RFC9260}}, DCCP {{!RFC4340}},
and QUIC {{!RFC9000}}.

A proposed congestion control algorithm SHOULD be evaluated when
competing with standard IETF congestion control {{!RFC5681}},
{{!RFC9260}}, {{!RFC4340}}, {{!RFC9002}}, {{!RFC9438}}.  A proposal
that has a significantly negative impact on
traffic using standard congestion control might be suspect and this aspect should
be part of the community's decision making with regards to the suitability of
the proposed congestion control algorithm. The community should also consider
other non-standard congestion control algorithms that are known to be widely deployed,

We note that this guideline is not a requirement for strict Reno- or Cubic-
friendliness as a prerequisite for an alternate congestion

> control mechanism to advance to Experimental status.  As an example,

Reviewer: ```suggestion
control mechanism to advance to Experimental or Standards Track status.  As an example,
```

extent they are not, experiments can be conducted where possible.

Note that in many deployments, real-time traffic is directed into distinct
queues via Differentiated Services Code Points (DSCP) or other mechanisms,
which substantially reduces the interplay with other traffic. However, a proposal
targeting Internet use
MUST NOT assume that all paths support specific mechanisms.

### Short and Long Flows

The effect on short-lived and long-lived flows using other common congestion
control algorithms MUST be evaluated, as in {{short-flows}}.

## Other Criteria

### Differences with Congestion Control Principles

Proposed congestion control algorithms SHOULD include a clear
explanation of any deviations from {{!RFC2914}} and {{!RFC7141}}.

### Incremental Deployment

The proposal ought to discuss whether the proposal
allows for incremental deployment in the
targeted environment.  For a mechanism targeted for deployment in
the current Internet, it would be helpful for a proposal to
discuss what is known (if anything) about the correct operation
of the mechanisms with some of the equipment that can be installed in the
current Internet, e.g., routers, transparent proxies, WAN

> optimizers, intrusion detection syms, home routers, and the

Reviewer: ```suggestion
optimizers, intrusion detection systems, home routers, and the
```

like.

As a similar concern, if the proposal
is intended only for specific environments (and not the
global Internet), the proposal should consider how this intention
is to be realised.  The community will have to address the
question of whether the scope can be enforced by stating

> a ste of restrictions or whether additional protocol mechanisms are

Reviewer 3: What does "a ste of" mean? Could this be a typo?

Author: whoops

At a minimum, a proposal should reason about an algorithm's response to various
AQM disciplines. Simulation or empirical results are, of course, valuable.

> Among the AQM techniques that might have an impact on a  proposed congestion control

Reviewer: ```suggestion
Among the AQM techniques that might have an impact on a proposed congestion control
```

algorithm are FQ-CoDel {{?RFC8290}}; Proportional Integral Controller Enhanced
(PIE) {{?RFC8033}}; and Low Latency, Low Loss, and Scalable Throughput (L4S)
{{?RFC9332}}.

Congestion control proposals that set one of the two Explicit Congestion Transport (ECT)
codepoints in the IP header can gain the benefits of receiving Explicit Congestion Notifictaion (ECN)
Congestion Experienced (CE) signals from an on-path AQM {{?RFC8087}}.
Use of ECN {{?RFC3168}},{{?RFC9332}} results in requirements for
the congestion control algorithm to react when it receives a packet with an ECN-CE marking.
This reaction needs to be evaluated to confirm that the algorithm conforms with the
requirements of the ECT codepoint that was used.

Note that evaluation of AQM techniques -- as opposed to their impact on specific

> proposals -- is out of scope of this document. {{?RFC7567}}

Reviewer: ```suggestion
congestion control proposals -- is out of scope of this document. {{?RFC7567}}
```

Variation also arises when a higher priority diffserv traffic classic prompts the
transmission by a lower class. In these cases, the delay varies as a function of
external factors and attempting to infer congestion from an increase in the delay
results in reduced throughput. The jitter from variation over short timescales
might not be distinguishable similar from other effects.

> Congetsion control proposals SHOULD be evaluated to ensure their operation is robust

Reviewer 3: ```suggestion
Congestion control proposals SHOULD be evaluated to ensure their operation is robust
```

## Internet of Things

The "Internet of Things" (IoT) is a broad concept, but when evaluating
a proposal,
it is often associated with unique characteristics.

IoT nodes might be more constrained in power, CPU, or other parameters than
conventional Internet hosts. This might place limits on the complexity of any
given algorithm. These power and radio constraints might make the volume of
control packets in a given algorithm a key evaluation metric.

Furthermore, many IoT applications do not a have a human in the loop, and
therefore can have weaker latency constraints because they do not relate to a user
experience, but still need to share the path with other flows with different
constraints.

Extremely low-power links can lead to very low throughput and a low bandwidth-
delay product, well below the standard operating range of most Internet flows.

## Paths with High Delay

A proposal ought not to presume that all general Internet paths have a low
delay.
Some paths include links that contibute much more delay than for a typical Internet path.
Satellite links often have delays longer than typical for wired paths
{{?RFC2488}} and high delay bandwidth products {{?RFC3649}}.

> Also, many syms

Reviewer: ```suggestion
Also, many systems
```

Even when the set of routers constituting a path does not change, the properties of
that path can vary with time (e.g., due to a change of link capacity, relative priority, or a change
in the rate of other traffic sharing a bottleneck), with a potential impact on the
operation of a congestion control algorithm.

## Multipath Transport

Multipath transport protocols permit more than one path to be differentiated and used by
a single connection at the sender.
A multipath sender can schedule which packets travel on which of its active paths.
This enables a tradeoff in timeliness and reliability. There are various ways that

> multipath techniques can be used.c

Reviewer 3: ```suggestion
multipath techniques can be used.
```

This has additional implications:
New CCs MUST evaluate the potential
harm to other flows when the multiple paths share a common

> conged bottleneck

Reviewer: ```suggestion
congested bottleneck
```

protocol suite and therefore does not directly impact Internet
security.  The implementation of various facets of the Internet's
current congestion control algorithms do have security implications
(e.g., as outlined in {{!RFC5681}}).

The IETF process that results in publication needs to

> ensure that these security implications are considreed.

Reviewer: ```suggestion
ensure that these security implications are considered.
```

Thanks to Richard Scheffenegger for helping to get this revision process started.

Discussions with Lars Eggert and Aaron Falk seeded the original RFC5033.

> Bob Briscoe, Gorry Fairhurst, Doug Leith, Jitendra Padhye, Christian Huetema,
> Matt Mathis,

Reviewer: ```suggestion
Bob Briscoe, Gorry Fairhurst, Doug Leith, Jitendra Padhye,
```

Colin Perkins, Pekka Savola, members of TSVWG, and participants at
the TCP Workshop at Microsoft Research all provided feedback and
contributions to that document.  It also drew from {{?RFC5166}}.

> Dave Taht also suggested improvements to this document.

Reviewer: ```suggestion

The editors would like to thanks to Reese Enghardt and Dave Taht for suggesting improvements to this document.
```
